common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -827,13 +833,16 @@ void resetChunkTracker() {
     void updateChunkInfo(long chunkOffset, int mapIndex) throws IOException {
       long idxStartPos = -1;
       try {
-        // update the chunk tracker to meta file before index file
-        writeChunkTracker(mapIndex);
         idxStartPos = indexFile.getFilePointer();
         logger.trace("{} shuffleId {} reduceId {} updated index current {} updated {}",
           appShuffleId.appId, appShuffleId.shuffleId, reduceId, this.lastChunkOffset,
           chunkOffset);
-        indexFile.writeLong(chunkOffset);
+        indexFile.write(Longs.toByteArray(chunkOffset));
+        // Chunk bitmap should be written to the meta file after the index file because if there are
+        // any exceptions during writing the offset to the index file, meta file should not be
+        // updated. If the update to the index file is successful but the update to meta file isn't
+        // then the index file position is reset in the catch clause.
```

Comment:
We don't need to update the `lastChunkOffset`. Since the current chunk offset was not successfully written to the index file, the `lastChunkOffset` would stay the same. The last chunk offset indicates the successfully written offset of the last chunk.

Reply:
Yeah..but the current implementation seems not correct. It still update the `lastChunkOffset` when exception happens, no?
https://github.com/apache/spark/blob/8113c88542ee282b510c7e046d64df1761a85d14/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java#L846

Reply:
If exception happens, line 853 would throw the exception skipping line 855.

Reply:
Ah, I see. Thanks!